### PR TITLE
feat(replay): Add docs for `mutationLimit`

### PR DESCRIPTION
**Pull request by billyvg**
Adds some documentation for `mutationLimit` and `mutationBreadcrumbLimit`

### CODE REVIEW

File: src/platforms/javascript/common/session-replay/configuration.mdx [hunk 1]
```diff
@@ -73,6 +75,17 @@ new Replay({
 
 <Note>Captured bodies will be truncated to 150k characters max.</Note>
 
+## Mutation Limits
+
+Session Replay works by recording incremental DOM changes that occur in your web application. Generally these changes occur in small batches and cause minimal overhead. However, it can be easy to overlook cases that will cause a large amount of DOM mutations to happen in a single update. An example is a custom dropdown component that has an unbounded list of options to render. This can negatively impact performance without Session Replay and its effects will be magnified with Session Replay enabled. In order to avoid this scenario, Session Replay will stop recording if it detects a large number of mutations (default: 10,000), which can be configured by setting `mutationLimit`. Additionally, we provide breadcrumbs in the replay to warn you when a large number of mutations are detected (750).
```
Comment (billyvg) [inline src/platforms/javascript/common/session-replay/configuration.mdx:80]: @bruno-garcia can you help me write up something in the performance overhead section of docs that links to this?

Was thinking we find some links talking about virtualization of tables, etc to improve performance, as a guide/example of the performance impacts of this normally (without replays)

**Comment by shanamatthews** [comment]
Someone from the docs team can take a pass at this once you add in the performance overhead info!

**Comment by billyvg** [comment]
@shanamatthews I think we can follow-up with a new PR for the Perf Overhead section

File: src/platforms/javascript/common/session-replay/configuration.mdx [hunk 2]
```diff
@@ -20,13 +20,15 @@ The following options can be configured on the root level of your browser-based
...
+| mutationLimit           | number               | 10000   | The upper bound of mutations to process before Session Replay stops recording due to performance impacts. See [Mutation Limits](#mutation-limits)             |
+| mutationBreadcrumbLimit | number               | 750     | The upper bound of mutations to process before Session Replay sends a breadcrumb to warn of large mutations. See [Mutation Limits](#mutation-limits)          |
+| networkDetailAllowUrls  | (string \| RegExp)[] | `[]`    | Capture request and response details for XHR and fetch requests that match the given URLs.                                                                    |
+| networkCaptureBodies    | boolean              | `true`  | Decide whether to capture request and response bodies for URLs defined in `networkDetailAllowUrls`.                                                           |
+| networkRequestHeaders   | string[]             | `[]`    | Capture request headers for URLs defined in `networkDetailAllowUrls`. See [Network Details](#network-details) to learn more.                                  |
```
Comment (vivianyentran) [inline src/platforms/javascript/common/session-replay/configuration.mdx:30]:
```suggestion
| networkRequestHeaders   | string[]             | `[]`    | Additional request headers to capture for URLs defined in `networkDetailAllowUrls`. See [Network Details](#network-details) to learn more.                                  |
```

File: src/platforms/javascript/common/session-replay/configuration.mdx [hunk 3]
```diff
@@ -20,13 +20,15 @@ The following options can be configured on the root level of your browser-based
...
+| networkResponseHeaders  | string[]             | `[]`    | Capture response headers for URLs defined in `networkDetailAllowUrls`. See [Network Details](#network-details) to learn more.                                 |
```
Comment (vivianyentran) [inline src/platforms/javascript/common/session-replay/configuration.mdx:31]:
```suggestion
| networkResponseHeaders  | string[]             | `[]`    | Additional response headers to capture for URLs defined in `networkDetailAllowUrls`. See [Network Details](#network-details) to learn more.                                 |
```

**Review by vivianyentran — APPROVED** [review]
Thanks for updating the docs! Added some suggestions to clarify the network header options.

**Review by bruno-garcia — APPROVED** [review]
Looks great! I'll add an entry pointing to this on a follow up PR

**Pull request merged**